PokemonViewer/DetailViewController.swift

```diff
@@ -0,0 +1,121 @@
+//
+//  DetailViewController.swift
+//  PokemonViewer
+//
+//  Created by Claire Chung on 2024/4/22.
+//
+
+import UIKit
+
+class DetailViewController: UIViewController {
+    // MARK: - Properties
+    var pokemon: Pokemon?
+    var nameLabel = UILabel()
+    var idLabel = UILabel()
+    var thumbnailImageView = UIImageView()
+    var descriptionLable = UILabel()
+    let descriptionIconImageView = UIImageView(
+        image: UIImage(systemName: "diamond.inset.filled")
+    )
+    let activityIndicator = UIActivityIndicatorView(style: .large)
+    private let networkManager: NetworkManager = .shared
+    
+    // MARK: - Methods
+    override func viewDidLoad() {
+        super.viewDidLoad()
+        view.backgroundColor = .white
+        
+        setupNameAndIdLabels()
+        setupImageView()
+        setupActivityIndicator()
+        
+        let urlString = "https://pokeapi.co/api/v2/pokemon-species/\(pokemon?.sprite?.id ?? 1)"
+        Task {
+            activityIndicator.startAnimating()
+            self.pokemon?.specy = await networkManager.fetchSpecy(from: urlString)
+            setupDescriptionLable()
+            activityIndicator.stopAnimating()
+        }
+    }
+    
+    private func setupNameAndIdLabels() {
+        let stackView = UIStackView(arrangedSubviews: [nameLabel, idLabel])
+        stackView.translatesAutoresizingMaskIntoConstraints = false
+        stackView.axis = .horizontal
+        stackView.distribution = .fill
+        stackView.alignment = .center
+        stackView.spacing = 20
+        
+        nameLabel.translatesAutoresizingMaskIntoConstraints = false
+        nameLabel.text = pokemon?.name
+        nameLabel.font = UIFont.systemFont(ofSize: 24, weight: .bold)
+        
+        idLabel.translatesAutoresizingMaskIntoConstraints = false
+        idLabel.text = "#" + String(format: "%04d", pokemon?.sprite?.id ?? 0 )
```

Comment:
通常建議不要用 `+` 的方式去拼組字串，那會用比較多效能去處理，雖然其實沒差但就是一個習慣，然後也不用用 String(format:) 去作轉換，通常只有在要把變數塞進 Localised 的字串中的時候會用到這個

```suggestion
        idLabel.text = "#\(pokemon?.sprite?.id ?? 0)"
```

Reply:
喔，後來想到你是要補 0，那 String(format:) 那可以保留